tests/plugins/clipboard/pasteimage.js

```diff
@@ -86,63 +21,107 @@
 			if ( !CKEDITOR.plugins.clipboard.isCustomDataTypesSupported ) {
 				assert.ignore();
 			}
-			FileReader.setFileMockType();
-			FileReader.setReadResult();
+
+			mockFileReader();
 			this.editor.focus();
 		},
 
+		tearDown: function() {
+			window.FileReader = originalFileReader;
+		},
+
 		'test paste .png from clipboard': function() {
 			FileReader.setFileMockType( 'image/png' );
 			FileReader.setReadResult( 'load' );
 
 			bender.tools.selection.setWithHtml( this.editor, '<p>Paste image here:{}</p>' );
-			this.assertPaste( 'image/png',
-				'<p>Paste image here:<img data-cke-saved-src="data:image/png;base64,fileMockBase64=" src="data:image/png;base64,fileMockBase64=" />^@</p>' );
+			this.assertPaste( {
+				type: 'image/png',
+				expected: '<p>Paste image here:<img data-cke-saved-src="data:image/png;base64,fileMockBase64=" src="data:image/png;base64,fileMockBase64=" />^@</p>'
+			} );
 		},
 
 		'test paste .jpeg from clipboard': function() {
 			FileReader.setFileMockType( 'image/jpeg' );
 			FileReader.setReadResult( 'load' );
 
 			bender.tools.selection.setWithHtml( this.editor, '<p>Paste image here:{}</p>' );
-			this.assertPaste( 'image/jpeg',
-				'<p>Paste image here:<img data-cke-saved-src="data:image/jpeg;base64,fileMockBase64=" src="data:image/jpeg;base64,fileMockBase64=" />^@</p>' );
+			this.assertPaste( {
+				type: 'image/jpeg',
+				expected: '<p>Paste image here:<img data-cke-saved-src="data:image/jpeg;base64,fileMockBase64=" src="data:image/jpeg;base64,fileMockBase64=" />^@</p>'
+			} );
 		},
 
 		'test paste .gif from clipboard': function() {
 			FileReader.setFileMockType( 'image/gif' );
 			FileReader.setReadResult( 'load' );
 
 			bender.tools.selection.setWithHtml( this.editor, '<p>Paste image here:{}</p>' );
-			this.assertPaste( 'image/gif',
-				'<p>Paste image here:<img data-cke-saved-src="data:image/gif;base64,fileMockBase64=" src="data:image/gif;base64,fileMockBase64=" />^@</p>' );
+			this.assertPaste( {
+				type: 'image/gif',
+				expected: '<p>Paste image here:<img data-cke-saved-src="data:image/gif;base64,fileMockBase64=" src="data:image/gif;base64,fileMockBase64=" />^@</p>'
+			} );
 		},
 
 		'test unsupported file type': function() {
 			FileReader.setFileMockType( 'application/pdf' );
 			FileReader.setReadResult( 'load' );
 
 			bender.tools.selection.setWithHtml( this.editor, '<p>Paste image here:{}</p>' );
-			this.assertPaste( 'application/pdf',
-				'<p>Paste image here:^@</p>' );
+			this.assertPaste( {
+				type: 'application/pdf',
+				expected: '<p>Paste image here:^@</p>'
+			} );
+		},
+
+		// (#4750)
+		'test showing notification for unsupported file type': function() {
```

Comment:
The same situation as above, let's add our expectations here:
```suggestion
		'test pasting unsupported file type shows notification': function() {
```